crates/miden-tx/src/host/mod.rs

```diff
@@ -317,8 +320,23 @@ where
         process: &mut ProcessState,
     ) -> Result<(), TransactionKernelError> {
         let pub_key = process.get_stack_word(0);
-        let msg = process.get_stack_word(1);
-        let signature_key = Hasher::merge(&[pub_key, msg]);
+        let message = process.get_stack_word(1);
+
+        let final_nonce = process
+            .get_mem_value(ContextId::root(), NATIVE_ACCT_ID_AND_NONCE_PTR + ACCT_NONCE_IDX as u32)
+            .unwrap();
+        let block_number = process
+            .get_mem_value(ContextId::root(), BLOCK_METADATA_PTR + BLOCK_NUMBER_IDX as u32)
+            .unwrap();
+
+        let salt = Word::from([ZERO, ZERO, block_number, final_nonce]);
+
+        let tx_summary = build_tx_summary(self, salt)
```

Comment:
I would potentially move the building of tx summary down to come after we verify that we have the signature (i.e., after the current line 345). Basically, the logic is as follows:

- Check if we have the signature.
- If we do, make sure the signature is over the transaction summary (which is a more expensive operation than checking that we have a signature).